lib/puppet/file_bucket/dipper.rb

```diff
@@ -56,6 +58,38 @@ def backup(file)
     end
   end
 
+  # Diffs two filebucket files identified by their sums
+  def diff(suma, sumb)
+    raise RuntimeError, "Diff is not supported on windows platforms" if Puppet.features.microsoft_windows?
+    source_path = "#{@rest_path}#{@checksum_type}/#{suma}"
+    if sumb and sumb != ''
+      file_diff = Puppet::FileBucket::File.indirection.find(
+        source_path,
+        :bucket_path => @local_path,
+        :diff_with => sumb)
+    else
+      original_paths = Puppet::FileBucket::File.indirection.find(
+        source_path,
+        :bucket_path => @local_path,
+        :original_path => true)
+      raise Puppet::Error, "Couldn't retrieve file's original path" unless original_paths
+      original_paths = original_paths.to_s.split(":")
+      file_diff = ""
+      original_paths.each do |original_path|
```

Comment:
I was actually suggesting not doing any of the magic for automatically diffing against where the file may be located on the current system and requiring the path to the file to diff against be passed like it would if you were diffing two hashes.  I prefer explicitness over magic, personally, but that doesn't mean we should go that route.  I just wanted to throw that suggestion out there as I think it gives a consistent "diff"-like interface to the command.